File: C#/1357 - Em Braille.cs
/// author Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
/// copyright Josélio Júnior (Lunatic Fox) 2022

using System;
using System.Linq;
using System.Collections.Generic;

class URI {
  static void Main(string[] args) { 
    string[] b = {
      ".***..", "*.....", "*.*...", "**....", "**.*..",
      "*..*..", "***...", "****..", "*.**..", ".**..."
    };

    while (true) {
      string r = Console.ReadLine() ?? "";
      if (r == "0") break;
      int n = int.Parse(r);
      string l = Console.ReadLine() ?? "";
  
      if (l == "B") {
        var sl = new List<List<string>>();
  
        for (int i = 0; i < 3; i++) {
        sl.Add((Console.ReadLine() ?? "").Split(' ').ToList());
        }
  
        var sm = sl.SelectMany(e => e.Select((ne, ni) => new { ne, ni }))
        .GroupBy(i => i.ni, i => i.ne)
        .Select(e => {
          string st = string.Concat(e);
          for (int i = 0; i < b.Length; i++)
          if (st == b[i]) st = i.ToString(); 
          return st;
        })
        .ToList();
  
        Console.WriteLine(string.Concat(sm));
      }
  
      if (l == "S") {
        var sl = (Console.ReadLine() ?? "")
        .ToCharArray()
        .Select(e => {
          string v = b[int.Parse(e.ToString())];
          var w = new List<string>();
  
          for (int i = 0; i < 3; i++)
          w.Add(string.Concat(v.Skip(i * 2).Take(2)));
  
          return w;
        })
        .SelectMany(e => e.Select((ne, ni) => new { ne, ni }))
        .GroupBy(i => i.ni, i => i.ne)
        .Select(e => string.Join(" ", e))
        .ToArray();
  
        Console.WriteLine(string.Join("\n", sl));
      }
    }
  }
}

File: Lua_5.4/1357 - Em Braille.lua
--- author: Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
--- copyright: Josélio Júnior (Lunatic Fox) 2023

local b = {
  '.***..', '*.....', '*.*...', '**....', '**.*..',
  '*..*..', '***...', '****..', '*.**..', '.**...'
}
while true do
  local n = tonumber(io.read())
  if n == 0 then break end
  if io.read() == 'B' then
    local t = {}
    for e in io.read():gmatch '[^ ]+' do table.insert(t, e) end
    for _ = 1, 2 do
      local j = 1
      for e in io.read():gmatch '[^ ]+' do
        t[j] = t[j] .. e
        j = j + 1
      end
    end
    for i = 1, #t do
      for j = 1, #b do
        if t[i] == b[j] then t[i] = j - 1 end
      end
    end
    print(table.concat(t))
  else
    local t = { {}, {}, {} }
    for e in io.read():gmatch '%d' do
      table.insert(t[1], b[e + 1]:sub(1, 2))
      table.insert(t[2], b[e + 1]:sub(3, 4))
      table.insert(t[3], b[e + 1]:sub(5, 6))
    end
    for i = 1, #t do t[i] = table.concat(t[i], ' ') end
    print(table.concat(t, '\n'))
  end
end
